test/python/transpiler/test_xy4.py

```diff
@@ -0,0 +1,133 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests XY4 dynamical decoupling pass."""
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.transpiler import PassManager
+from qiskit.transpiler.passmanager_config import PassManagerConfig
+from qiskit.transpiler.passes import XY4
+from qiskit.test import QiskitTestCase
+from qiskit.test.mock import FakeAlmaden
+
+class TestXY4(QiskitTestCase):
+    """Test the XY4 DD pass."""
+
+    def setUp(self):
```

Comment:
Is there a test that verifies the behaviour of qubits that are not used in the circuit? What is the behaviour?

Reply:
Good idea. It shouldn't do anything on those qubits (and I don't believe it will, given the check on line  127 `if seq_duration > delay_duration or len(dag.ancestors(node)) <= 1:`)